src/lib.rs

```diff
@@ -638,4 +654,105 @@ impl PathFind {
 fn sc2pathlib(_py: Python, m: &PyModule) -> PyResult<()> {
     m.add_class::<PathFind>()?;
     Ok(())
-}
\ No newline at end of file
+}
+#[cfg(test)]
+mod tests{
+    use super::*;
+    use test::bench::Bencher;
+    use std::fs::File;
+    use std::io::{BufRead, BufReader};
+
+    fn rotate90clockwise(vec: Vec<Vec<usize>>) -> Vec<Vec<usize>>{
+        let N = vec[0].len();
+        let mut new_arr: Vec<Vec<usize>>= vec.clone();
+        // Traverse each cycle
+        for i in 0..(N / 2) {
+            for j in i..(N-i-1){
+                let temp = vec[i][j];
+                new_arr[i][j] = vec[N-1-j][i];
+                new_arr[N-1-j][i] = vec[N-1-i][N-1-j];
+                new_arr[N-1-i][N-1-j]= vec[j][N-1-i];
+                new_arr[j][N-1-j] = temp;
+
+            }
+
+        }
+        new_arr
+    }
+
+    fn read_vec_from_file(file_path: &str)-> Vec<Vec<usize>>{
+
+        let f = BufReader::new(File::open(file_path).unwrap());
+        let mut arr = Vec::<Vec<usize>>::new();
+
+        for line in f.lines().map(|x| x.unwrap()){
+            let mut maze_line = vec![];
+            for mini_line in line.chars().map(|n| n.to_digit(2).unwrap()){
+                maze_line.push(mini_line as usize)
+            }
+
+            arr.push(maze_line);
+        }
+        rotate90clockwise(arr)
+
+    }
+
+    fn get_pathfind(file: &str)->PathFind{
+        let map = read_vec_from_file(file);
+        PathFind::bench_new(map)
+    }
+
+    #[test]
+    fn test_find_path_automaton_le(){
+        let path_find = get_pathfind("tests/AutomatonLE.txt");
+        let r = path_find.find_path((32, 51),(150, 118),Some(0));
+        let (_, distance) = r.unwrap();
+        assert!(distance > 0_f64);
+    }
+
+    #[test]
+    fn test_find_path_4x4(){
+        let path_find = get_pathfind("tests/maze4x4.txt");
+        let r = path_find.find_path((0, 0), (3, 3),Some(0));
+        let (_, distance) = r.unwrap();
+        assert!(distance > 0_f64);
```

Comment:
Can we have the test compare the actual to expected result? that way we'll know if something changes. If that is too much work for now ti figure them out, perhaps add a TODO comment instead?

Reply:
I've been thinking about this. Maybe it's best to save the expected result in a file, and load that into a variable then? I think it will make the code look cluttered if we save the entire expected result in the code itself. Thoughts?

Reply:
I don't see a problem with having the expected result be part of the test. It's a pain to start searching for the "correct result" in case something goes wrong. In NUnit in C# there was nice way to set the parameters as test attributes. Like you could have something like this
```cs
[Test(10, "testmap4x4", (0,0), (0,2)]
void Test(float expected, string map, Point2 start, Point2 end)
    // ...
```